analyzer/codechecker_analyzer/cmd/check.py

```diff
@@ -356,13 +356,17 @@ def add_arguments_to_parser(parser):
     analyzer_opts.add_argument('--analyzer-config',
                                dest='analyzer_config',
                                nargs='*',
-                               default=argparse.SUPPRESS,
+                               default=["clang-tidy:HeaderFilterRegex=.*"],
                                help="Analyzer configuration options in the "
                                     "forllowing format: analyzer:key=value. "
                                     "The collection of the options can be "
                                     "printed with "
                                     "'CodeChecker analyzers "
-                                    "--analyzer-config'.")
+                                    "--analyzer-config'. To disable the "
```

Comment:
Wouldn't be the disable of the default behavior be something like this `--analyzer-config clang-tidy:""` an empty config here. The default behavior here from the codechecker point of view is this `--analyzer-config clang-tidy:"clang-tidy:HeaderFilterRegex=.*"` right?

It the config is taken from a clang tidy config file the analyzer and checker config values are mixed there together.
Where should the user enable the config file usage as an `--analyzer-config` or as a `--checker-config`?
I think the analyzer configuration documentation should be extended with examples what can be configured and how to do it and it would be nice to mention here that the clang-tidy default behavior is to search for a .clang-tidy config file.
So it will be clear to the how to use the configuration options.

Reply:
This `take-config-from-directory` config option has been introduced by us so it is explicitly visible in the analyze command that the `.clang-tidy` file is being used.

`clang-tidy` doesn't mix the content of `.clang-tidy` config file and the command line config options. We also preserve this approach by not mixing and only one of these will be taken into account.